src/crawlers/cheerio_crawler.js

```diff
@@ -403,8 +406,17 @@ class CheerioCrawler {
         // Using the streaming API of Request to be able to
         // handle the response based on headers receieved.
         const opts = this._getRequestOptions(request);
-
-        const responseStream = await requestAsBrowser(opts);
+        let responseStream;
+
+        try {
+            responseStream = await requestAsBrowser(opts);
+        } catch (e) {
+            if (e instanceof TimeoutError) {
+                throw new Error(TIMEOUT_ERROR_MESSAGE);
```

Comment:
Does this really throw here when it's in stream mode? The `await` returns immediately and the timeout hits "some time later".